File: notification-service/src/main/java/com/eirsteir/coffeewithme/notification/domain/Notification.java
package com.eirsteir.coffeewithme.notification.domain;

import com.eirsteir.coffeewithme.commons.domain.notification.NotificationType;
import com.fasterxml.jackson.annotation.JsonIgnore;
import java.util.Date;
import javax.persistence.*;
import lombok.*;
import lombok.experimental.Accessors;
import org.hibernate.annotations.CreationTimestamp;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Accessors(chain = true)
@Entity
public class Notification {

  @Id
  @GeneratedValue(strategy = GenerationType.AUTO)
  private Long notificationId;

  @CreationTimestamp private Date timestamp;

  private NotificationType type; // TODO: 21.05.2020 Add converter

  @Builder.Default private boolean seen = false;

  @Embedded private UserDetails user;

  @JsonIgnore private Long subjectId;

  @Getter
  @Builder
  @NoArgsConstructor
  @AllArgsConstructor
  @Embeddable
  public static class UserDetails {
    private Long id;
    private String name;
    private String nickname;
  }
}

File: notification-service/src/main/kotlin/com/eirsteir/coffeewithme/notification/domain/Notification.kt
package com.eirsteir.coffeewithme.notification.domain

import com.eirsteir.coffeewithme.commons.domain.notification.NotificationType
import com.fasterxml.jackson.annotation.JsonIgnore
import lombok.*
import lombok.experimental.Accessors
import org.hibernate.annotations.CreationTimestamp
import java.util.*
import javax.persistence.*


@Entity
data class Notification(
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    val notificationId: Long? = null,
    @CreationTimestamp
    val timestamp: Date? = null,
    val type : NotificationType? = null, // TODO: 21.05.2020 Add converter
    var seen: Boolean = false,
    @Embedded
    val user: UserDetails? = null,
    @JsonIgnore
    val subjectId: Long? = null,
) {
    @Embeddable
    data class UserDetails(
        var id: Long? = null,
        var name: String? = null,
        var nickname: String? = null,
    )
}